src/Operators/derivative_operators.jl

```diff
@@ -94,3 +94,27 @@
 @inline ∂yᶠᶠᵃ(i, j, k, grid, u, args...) = δyᵃᶠᵃ(i, j, k, grid, u, args...) / Δyᶠᶠᵃ(i, j, k, grid)
 @inline ∂yᶜᶠᵃ(i, j, k, grid, c, args...) = δyᵃᶠᵃ(i, j, k, grid, c, args...) / Δyᶜᶠᵃ(i, j, k, grid)
 
+#####
+##### Given nothing the derivative is zero
+#####
+
+@inline ∂xᶜᵃᵃ(i, j, k, grid, ::Nothing, args...) = 0
+@inline ∂xᶠᵃᵃ(i, j, k, grid, ::Nothing, args...) = 0
+
+@inline ∂xᶜᶜᵃ(i, j, k, grid, ::Nothing, args...) = 0
+@inline ∂xᶜᶠᵃ(i, j, k, grid, ::Nothing, args...) = 0
+
+@inline ∂xᶠᶠᵃ(i, j, k, grid, ::Nothing, args...) = 0
+@inline ∂xᶠᶜᵃ(i, j, k, grid, ::Nothing, args...) = 0
+
+@inline ∂yᵃᶜᵃ(i, j, k, grid, ::Nothing, args...) = 0
+@inline ∂yᵃᶠᵃ(i, j, k, grid, ::Nothing, args...) = 0
+
+@inline ∂yᶜᶜᵃ(i, j, k, grid, ::Nothing, args...) = 0
+@inline ∂yᶠᶜᵃ(i, j, k, grid, ::Nothing, args...) = 0
+
+@inline ∂yᶠᶠᵃ(i, j, k, grid, ::Nothing, args...) = 0
+@inline ∂yᶜᶠᵃ(i, j, k, grid, ::Nothing, args...) = 0
+
+@inline ∂zᵃᵃᶜ(i, j, k, grid, ::Nothing, args...) = 0
+@inline ∂zᵃᵃᶠ(i, j, k, grid, ::Nothing, args...) = 0
```

Comment:
Another solution (slightly less code) is to define a method like `hydrostatic_pressure_gradient_x(i, j, k, grid, p)`.

Then `p::Nothing` returns 0 and otherwise returns `∂xᶠᶜᵃ(i, j, k, grid, p)`.

What do you think?

Reply:
That's  much better I think.

I was also a bit skeptical for `∂x` to return 0 when given nothing. We may not always want that.